scipy/stats/_stats_py.py

```diff
@@ -4780,8 +4794,11 @@ def statistic(x, y, axis):
     pvalue = _get_pvalue(r, dist, alternative, xp=xp)
 
     mask = (n == 2)   #  return exactly 1.0 or -1.0 values for n == 2 case as promised
-    def special_case(r): return xp.where(xp.isnan(r), xp.nan, xp.ones_like(r))
-    r = xpx.apply_where(mask, (r,), xp.round, fill_value=r)
+    # data-apis/array-api-extra#196
+    mxp = array_namespace(r._meta) if is_dask(xp) else xp    
+    def special_case(r):
+        return mxp.where(mxp.isnan(r), mxp.nan, mxp.ones_like(r))
```

Comment:
To check my understanding, an alternative fix would be for `special_case` to detect the `xp` of `r`?
```suggestion
    def special_case(r):
        xp = array_namespace(r)
        return mxp.where(xp.isnan(r), xp.nan, xp.ones_like(r))
```

Reply:
I see.

I think I'm a little confused by the terminology, though. I think `m` in `mxp` refers to `meta`, since it comes from `r._meta`. But take the case of Dask with NumPy as the underlying library - if we are to use the term "meta" to refer to one of the libraries, I would expect it to refer to the higher-level library, Dask, not NumPy. But the callable passed to `apply_where` needs to work on arrays from underlying library (NumPy). Is the use of "meta" reversed compared to what I would expect?

Reply:
Dask has used the word "meta" for almost a decade to indicate the wrapped namespace.
I would advise against overloading it now to mean the opposite in some cases.
"wrapper" and "wrapped" are more explicit, but unfortunately don't abbreviate easily.

Reply:
I did a little more research, and now understand that `_meta` is used in the sense of "metadata". I don't think Dask is referring to NumPy as the "meta" namespace. The `_meta` attribute stores metadata about what a chunk will look like, and it happens to be an instance of the underlying library's array type, so we are able to use `array_namespace` on it to get the underlying library.